webapp/src/features/rsuUpdateMenu/rsuUpdateMenuSlice.js

```diff
@@ -0,0 +1,137 @@
+      try {
+        const data = await apiHelper._postData({
+          url: EnvironmentVars.rsuRestAuth,
+          token,
+          body: JSON.stringify({
+            command: "checkforupdates",
+            rsu_ip: ipList[i]["properties"]["Ipv4Address"],
+            args: {},
+          }),
+        }).body;
+
+        if (data.os === true) {
+          osUpdateAvailableList.push(ipList[i].properties.Ipv4Address);
+        }
+        if (data.firmware === true) {
+          fwUpdateAvailableList.push(ipList[i].properties.Ipv4Address);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    return { osUpdateAvailableList, fwUpdateAvailableList };
+  },
+  { condition: (_, { getState }) => selectToken(getState()) }
+);
+
+export const performOSUpdate = createAsyncThunk(
+  "rsuUpdateMenu/performOSUpdate",
+  async (ip, { getState }) => {
+    const currentState = getState();
+    const token = selectToken(currentState);
+
+    const data = await apiHelper._postData({
+      url: EnvironmentVars.rsuRestAuth,
+      token,
+      body: JSON.stringify({
+        command: "osupdate",
+        rsu_ip: ip,
+        args: {},
+      }),
+    });
+
+    switch (data.status) {
+      case 200:
+        console.debug(data.body);
+        return;
+      default:
+        console.error(data.message);
+        return;
+    }
+  },
+  { condition: (_, { getState }) => selectToken(getState()) }
+);
+
+export const performFWUpdate = createAsyncThunk(
+  "rsuUpdateMenu/performFWUpdate",
+  async (ip, { getState }) => {
+    const currentState = getState();
+    const token = selectToken(currentState);
+
+    const data = await apiHelper._postData({
+      url: EnvironmentVars.rsuRestAuth,
+      token,
+      body: JSON.stringify({
+        command: "fwupdate",
+        rsu_ip: ip,
+        args: {},
+      }),
+    });
+
+    switch (data.status) {
+      case 200:
+        console.log(data.body);
+        return;
+      default:
+        console.error(data.message);
+        return;
+    }
+  },
+  { condition: (_, { getState }) => selectToken(getState()) }
+);
+
+export const rsuUpdateMenuSlice = createSlice({
+  name: "rsuUpdateMenu",
+  initialState: {
+    loading: false,
+    value: initialState,
+  },
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(updateRsuData.pending, (state) => {
+        state.loading = true;
+      })
+      .addCase(updateRsuData.fulfilled, (state, action) => {
+        state.loading = false;
+        state.value.checked = true;
+        state.value.osUpdateAvailable = [...state.value.osUpdateAvailable, action.payload.osUpdateAvailableList];
+        state.value.fwUpdateAvailable = [...state.value.fwUpdateAvailable, action.payload.fwUpdateAvailableList];
+      })
+      .addCase(updateRsuData.rejected, (state) => {
+        state.loading = false;
+      });
+  },
+});
+
+export const {} = rsuUpdateMenuSlice.actions;
```

Comment:
Does this empty object somehow just export all actions?

Reply:
There are no reducers defined here, so this is technically not necessary. However, it matches the other slices, and makes it easier to add new reducers in the future